core/src/main/java/ai/timefold/solver/core/impl/localsearch/decider/reconfiguration/AdaptedSolverScope.java

```diff
@@ -0,0 +1,317 @@
+package ai.timefold.solver.core.impl.localsearch.decider.reconfiguration;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicReference;
+
+import ai.timefold.solver.core.api.score.Score;
+import ai.timefold.solver.core.api.solver.ProblemSizeStatistics;
+import ai.timefold.solver.core.config.solver.monitoring.SolverMetric;
+import ai.timefold.solver.core.impl.domain.solution.descriptor.SolutionDescriptor;
+import ai.timefold.solver.core.impl.localsearch.decider.LocalSearchDecider;
+import ai.timefold.solver.core.impl.score.definition.ScoreDefinition;
+import ai.timefold.solver.core.impl.score.director.InnerScoreDirector;
+import ai.timefold.solver.core.impl.solver.AbstractSolver;
+import ai.timefold.solver.core.impl.solver.change.DefaultProblemChangeDirector;
+import ai.timefold.solver.core.impl.solver.scope.SolverScope;
+import ai.timefold.solver.core.impl.solver.thread.ChildThreadType;
+
+import io.micrometer.core.instrument.Tags;
+
+public class AdaptedSolverScope<Solution_> extends SolverScope<Solution_> {
```

Comment:
This is _a lot_ of code just to pass in one extra parameter. It also creates uncertainty as to what type exactly is the solver scope, and who set that type there. 

Since solver scope is mutable and is already a "god class" used for storing whatever context we have, why not make the decider just another field on the original solver scope? Arguably though, it should not be on the solver scope, because it's the `localSearchDecider` - it should be on local search's phase scope.